### PR TITLE
Refactor parquet::arrow module

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 
As the reader logic in particular has gotten more sophisticated, the module has gotten rather large. This moves some files around to make the structure easier to understand

# What changes are included in this PR?

* Creates a new buffer module where the arrow compat logic can live
* Renames files to `module_name/mod.rs` instead of having `module_name.rs` and `module_name/` 

# Are there any user-facing changes?

No

### CODE REVIEW

File: parquet/src/arrow/buffer/mod.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Logic for reading data into arrow buffers
+
+pub mod bit_util;
+pub mod offset_buffer;
+pub mod dictionary_buffer;
+pub mod converter;
```
Comment (tustvold) [inline parquet/src/arrow/buffer/mod.rs:23]: I would like to eventually upstream many of these into arrow-rs proper

File: parquet/benches/arrow_reader.rs [hunk 2]
```diff
@@ -355,27 +355,6 @@ fn create_string_byte_array_dictionary_reader(
     .unwrap()
 }
 
-fn create_complex_object_byte_array_dictionary_reader(
```
Comment (tustvold) [inline parquet/benches/arrow_reader.rs:358]: ComplexObjectArrayReader is no longer used and is slated for removal (#1661 )

**Review by Ted-Jiang — COMMENTED** [review]
>As the reader logic in particular has gotten more sophisticated
Great !👍  I took me a lot of time to figure out how this work 😂

**Comment by tustvold** [comment]
> Is there any plan for simplify the reader API

Yes, the high level ticket is #1163. I'm hoping to pick it up soon, but large amounts of this API are public so needs some careful thought

**Comment by Ted-Jiang** [comment]
> Yes, the high level ticket is #1163. I'm hoping to pick it up soon, but large amounts of this API are public so needs some careful thought

Great! 👍 After implement send row_selection in `arrow_reader::get_record_reader_by_columns`
I would glad to help! 😊

**Pull request merged**